docs/devel/kubectl-conventions.md

```diff
@@ -58,19 +60,15 @@ Updated: 8/27/2015
 ## Command conventions
 
 * Command names are all lowercase, and hyphenated if multiple words.
-* kubectl VERB NOUNs for commands that apply to multiple resource types
-* NOUNs may be specified as TYPE name1 name2 ... or TYPE/name1 TYPE/name2; TYPE is omitted when only a single type is expected
-* Resource types are all lowercase, with no hyphens; both singular and plural forms are accepted
+* kubectl VERB NOUNs for commands that apply to multiple resource types.
+* NOUNs may be specified as TYPE name1 name2 ... or TYPE/name1 TYPE/name2; TYPE is omitted when only a single type is expected.
+* Resource types are all lowercase, with no hyphens; both singular and plural forms are accepted.
 * NOUNs may also be specified by one or more file arguments: -f file1 -f file2 ...
 * Resource types may have 2- or 3-letter aliases.
 * Business logic should be decoupled from the command framework, so that it can be reused independently of kubectl, cobra, etc.
-  * Ideally, commonly needed functionality would be implemented server-side in order to avoid problems typical of "fat" clients and to make it readily available to non-Go clients
-* Commands that generate resources, such as `run` or `expose`, should obey the following conventions:
-  * Flags should be converted to a parameter Go map or json map prior to invoking the generator
-  * The generator must be versioned so that users depending on a specific behavior may pin to that version, via `--generator=`
-  * Generation should be decoupled from creation
-  * `--dry-run` should output the resource that would be created, without creating it
-* A command group (e.g., `kubectl config`) may be used to group related non-standard commands, such as custom generators, mutations, and computations
+  * Ideally, commonly needed functionality would be implemented server-side in order to avoid problems typical of "fat" clients and to make it readily available to non-Go clients.
+* Commands that generate resources, such as `run` or `expose`, should obey specific conventions, see [generators](#generators).
+* A command group (e.g., `kubectl config`) may be used to group related non-standard commands, such as custom generators, mutations, and computations.
```

Comment:
We don't have agreed-upon principles about when a command group that doesn't start with a verb is appropriate vs. not.